packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -24,15 +28,66 @@ import {
 	type VisualTreeNode,
 } from "./VisualTree.js";
 
+/**
+ * Returns allowed types of the non-leaf nodes in the tree.
+ * - String: Allowed type for the array node.
+ * - `Record<string, string>`: Allowed types for the non-array nodes (e.g., object, map).
```

Comment:
I'm not sure I understand this part of the comment. This function returns a `VisualChildNode`, not a string or record.